### PR TITLE
feat(extract): implement configurable connection pattern DSL with decorator matching (#242)

### LINKED ISSUE #242: [M4-D4.1/D4.2] Implement configurable connection pattern DSL with decorator matching
### Context
**PRD:** \`docs/project/PRD/active/PRD-phase-12-connection-detection.md\` — M4 D4.1, D4.2

Teams with existing conventions need configurable pattern matching. The DSL supports methodCalls finder with where clauses and extract rules, plus decorator-based matching for frameworks like NestJS. D4.3 (interface mappings) was removed per §9.5 Q3.

### Traceability
- **PRD Section:** M4-D4.1 (Core connection pattern DSL), M4-D4.2 (Decorator-based matching)
- **Functional Requirements:** methodCalls finder, where clauses, extract rules, decorator matching
- **Success Criteria:** SC-5 (Configurable layer supports custom patterns via DSL)

### Acceptance Criteria
- ✓ ConnectionPattern type defined in riviere-extract-config with JSON Schema validation
- ✓ \`find: methodCalls\` finder with \`where\` clauses: methodName, receiverType
- ✓ \`extract\` rules: fromArgument (N), fromReceiverType, fromCallerType
- ✓ \`linkType\`: sync or async
- ✓ \`callerHasDecorator\` clause — array of decorator names
- ✓ \`calleeType.hasDecorator\` clause — decorator name on the callee's class
- ✓ Decorator matching is name-only: @Controller('/orders') matches "Controller". Parameters ignored.
- ✓ Composed and factory decorators NOT resolved — only direct decorators matched
- ✓ fromArgument: N — resolves static type of argument at position N, reads named property from type
- ✓ fromReceiverType — extracts static type name of called object
- ✓ fromCallerType — extracts static type name of class containing call site
- ✓ Connection config at global level (connections.patterns) and per-module level
- ✓ Module-level patterns additive to global patterns
- ✓ Config schema added to riviere-extract-config with JSON Schema validation
- ✓ 100% branch coverage

### Edge Case Scenarios
- Pattern with methodName that matches no calls → no links produced (not an error)
- Pattern with receiverType that matches no types → no links produced
- fromArgument with argument index out of bounds → strict fails, lenient marks uncertain
- fromArgument where argument type cannot be statically resolved → strict fails, lenient marks uncertain
- fromArgument where resolved type lacks the extract property → strict fails, lenient marks uncertain
- Decorator with parentheses @Controller() vs without @Controller → both match "Controller"
- Multiple patterns matching same call site → one link per pattern match (may produce duplicates, deduplicated by unique tuple)
- Empty patterns array → no configurable detection, Golden Path only
- Global pattern + module pattern both match same call → deduplicated

### Implementation Guidelines
- **Architecture:** New \`domain/connection-detection/configurable/\` in \`riviere-extract-ts\` per §9.1.2. New \`connections\` config key in \`riviere-extract-config\` per §9.1.3.
- **Firm constraints:** §9.2.5 (ConnectionPattern value object), §9.1.3 (global + per-module inheritance)
- **Flexible decisions:** Implementation of pattern evaluation engine
- ConnectionPattern type in extract-config: \`{ name, find, where, extract, linkType }\`
- Reuses ComponentIndex and call graph infrastructure from D1.1
- Follow existing config validation patterns in \`packages/riviere-extract-config/src/validation.ts\`
- Decorator detection via ts-morph \`getDecorators()\` on class declarations

### Embedded Reasoning
**Why:** Not all teams use Golden Path conventions. Configurable patterns let teams extract connections from existing codebases with different conventions (NestJS, Express, custom event emitters).
**What:** ConnectionPattern type + JSON Schema, pattern evaluation engine matching methodCalls with where/extract clauses, decorator-based matching.
**How:** Define ConnectionPattern in extract-config. In extract-ts, evaluate each pattern against call sites found by the call graph. Match where clauses, apply extract rules, produce ExtractedLink[].

### Testing Strategy
- **Unit:** ConnectionPattern schema validation (valid/invalid configs). Pattern evaluation: methodName matching, receiverType matching, extract rules (fromArgument, fromReceiverType, fromCallerType). Decorator matching: name-only, with/without params, composed decorators ignored.
- **Integration:** NestJS-style test fixtures with @Controller/@Injectable producing correct links. Custom event emitter patterns.
- **Edge Cases:** Out-of-bounds argument index, unresolvable types, empty patterns, global + module pattern interaction.

### Dependencies
Depends on #237

### Verification
\`\`\`bash
pnpm nx test riviere-extract-config -- --testNamePattern "connection"
pnpm nx test riviere-extract-ts -- --testNamePattern "configurable"
pnpm nx test riviere-extract-ts -- --coverage
\`\`\`

### PR DESCRIPTION
**Pull request by NTCoding**
Closes #242

### Context
**PRD:** \`docs/project/PRD/active/PRD-phase-12-connection-detection.md\` — M4 D4.1, D4.2

Teams with existing conventions need configurable pattern matching. The DSL supports methodCalls finder with where clauses and extract rules, plus decorator-based matching for frameworks like NestJS. D4.3 (interface mappings) was removed per §9.5 Q3.

### Traceability
- **PRD Section:** M4-D4.1 (Core connection pattern DSL), M4-D4.2 (Decorator-based matching)
- **Functional Requirements:** methodCalls finder, where clauses, extract rules, decorator matching
- **Success Criteria:** SC-5 (Configurable layer supports custom patterns via DSL)

### Acceptance Criteria
- ✓ ConnectionPattern type defined in riviere-extract-config with JSON Schema validation
- ✓ \`find: methodCalls\` finder with \`where\` clauses: methodName, receiverType
- ✓ \`extract\` rules: fromArgument (N), fromReceiverType, fromCallerType
- ✓ \`linkType\`: sync or async
- ✓ \`callerHasDecorator\` clause — array of decorator names
- ✓ \`calleeType.hasDecorator\` clause — decorator name on the callee's class
- ✓ Decorator matching is name-only: @Controller('/orders') matches "Controller". Parameters ignored.
- ✓ Composed and factory decorators NOT resolved — only direct decorators matched
- ✓ fromArgument: N — resolves static type of argument at position N, reads named property from type
- ✓ fromReceiverType — extracts static type name of called object
- ✓ fromCallerType — extracts static type name of class containing call site
- ✓ Connection config at global level (connections.patterns) and per-module level
- ✓ Module-level patterns additive to global patterns
- ✓ Config schema added to riviere-extract-config with JSON Schema validation
- ✓ 100% branch coverage

### Edge Case Scenarios
- Pattern with methodName that matches no calls → no links produced (not an error)
- Pattern with receiverType that matches no types → no links produced
- fromArgument with argument index out of bounds → strict fails, lenient marks uncertain
- fromArgument where argument type cannot be statically resolved → strict fails, lenient marks uncertain
- fromArgument where resolved type lacks the extract property → strict fails, lenient marks uncertain
- Decorator with parentheses @Controller() vs without @Controller → both match "Controller"
- Multiple patterns matching same call site → one link per pattern match (may produce duplicates, deduplicated by unique tuple)
- Empty patterns array → no configurable detection, Golden Path only
- Global pattern + module pattern both match same call → deduplicated

### Implementation Guidelines
- **Architecture:** New \`domain/connection-detection/configurable/\` in \`riviere-extract-ts\` per §9.1.2. New \`connections\` config key in \`riviere-extract-config\` per §9.1.3.
- **Firm constraints:** §9.2.5 (ConnectionPattern value object), §9.1.3 (global + per-module inheritance)
- **Flexible decisions:** Implementation of pattern evaluation engine
- ConnectionPattern type in extract-config: \`{ name, find, where, extract, linkType }\`
- Reuses ComponentIndex and call graph infrastructure from D1.1
- Follow existing config validation patterns in \`packages/riviere-extract-config/src/validation.ts\`
- Decorator detection via ts-morph \`getDecorators()\` on class declarations

### Embedded Reasoning
**Why:** Not all teams use Golden Path conventions. Configurable patterns let teams extract connections from existing codebases with different conventions (NestJS, Express, custom event emitters).
**What:** ConnectionPattern type + JSON Schema, pattern evaluation engine matching methodCalls with where/extract clauses, decorator-based matching.
**How:** Define ConnectionPattern in extract-config. In extract-ts, evaluate each pattern against call sites found by the call graph. Match where clauses, apply extract rules, produce ExtractedLink[].

### Testing Strategy
- **Unit:** ConnectionPattern schema validation (valid/invalid configs). Pattern evaluation: methodName matching, receiverType matching, extract rules (fromArgument, fromReceiverType, fromCallerType). Decorator matching: name-only, with/without params, composed decorators ignored.
- **Integration:** NestJS-style test fixtures with @Controller/@Injectable producing correct links. Custom event emitter patterns.
- **Edge Cases:** Out-of-bounds argument index, unresolvable types, empty patterns, global + module pattern interaction.

### Dependencies
Depends on #237

### Verification
\`\`\`bash
pnpm nx test riviere-extract-config -- --testNamePattern "connection"
pnpm nx test riviere-extract-ts -- --testNamePattern "configurable"
pnpm nx test riviere-extract-ts -- --coverage
\`\`\`

## Summary by CodeRabbit

* **New Features**
  * Pattern-driven connection detection (configurable connections) and a consolidated extraction command with automatic repository detection; optional timing output via --stats.

* **Bug Fixes**
  * Stricter validations and clearer failures (min-length checks, explicit errors instead of silent fallbacks); improved CLI error handling and safer test parsing/assertions.

* **Tests**
  * Major expansion of unit and integration tests covering connection detection, extraction flows, validation, and error handling.

### CODE REVIEW

File: packages/riviere-cli/src/features/extract/commands/run-extraction.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        type: c.type,
+        name: c.name,
+        domain: c.domain,
+      })),
+      modified: [],
+      removed: [],
+    })
+    console.log(markdown)
+    return
+  }
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply packages/riviere-cli/src/features/extract/commands/run-extraction.ts]: ❌ **Rejected**: The categorizeComponents function at cli-presentation/categorize-components.ts was NOT modified in this PR. It is an existing function that was introduced in a previous PR for the --format markdown feature. The feedback requests implementing a full change detection system (loading baseline graphs, comparing extracted components, producing added/modified/removed categorization). This is future PRD work that requires a baseline graph infrastructure which does not exist yet. The current behavior correctly shows all extracted components from changed files in a PR — the categorization is about the extraction result, not about what changed in the graph. This is outside the scope of the current PR which focuses on error handling refactoring and CodeRabbit feedback fixes.

---
*Generated by Claude Code*

File: packages/riviere-cli/src/features/extract/entrypoint/extract.pr-extraction.spec.ts [hunk 2]
```diff
@@ -17,6 +17,14 @@ import {
   createValidExtractFixture,
 } from '../__fixtures__/extraction-test-fixtures'
 
+vi.mock('../../../platform/infra/git/git-repository-info', () => ({
+  getRepositoryInfo: vi.fn(() => ({
+    name: 'test/repo',
+    owner: 'test',
+    url: 'https://github.com/test/repo.git',
+  })),
+}))
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply packages/riviere-cli/src/features/extract/entrypoint/extract.pr-extraction.spec.ts]: ❌ **Rejected**: This duplication is intentional and documented. Vitest requires vi.mock() to be called at the top level of each test file — it cannot be extracted to a shared fixture function and called dynamically. The vi.mock() hoisting mechanism (vi.mock is hoisted to the top of the file at compile time) means the mock path and factory must be statically analyzable in each file. See https://vitest.dev/api/vi.html#vi-mock — the factory function is hoisted and cannot be imported from another module. The extraction-test-fixtures.ts file already contains a comment at lines 107-109 documenting this constraint: 'vi.mock() for git-repository-info is duplicated across spec files because vi.mock() must be called at the top level with an inline factory.' Each mock is only 5 lines and the duplication is the correct pattern for Vitest.

---
*Generated by Claude Code*
